cruise.umple/test/cruise/umple/compiler/RuleInstanceTest.java

```diff
@@ -7,486 +7,486 @@
-    Assert.assertEquals(0,instance.getRulePart(1).getNextIdentifiers().length);
-    
-    instance.updateNextIdentifier();
-    Assert.assertEquals(1,instance.getRulePart(1).getNextIdentifiers().length);
-    Assert.assertEquals(";",instance.getRulePart(1).getNextIdentifiers()[0]);
-  }  
-
-  @Test
-  public void getNextIdentifier_Complex()
-  {
-    parser.addRule("attribute : [=unique]? [=modifier:immutable|settable|internal|defaulted|const]? [type,name>1,0] (= [value])? ;");
-    RuleInstance instance = new RuleInstance(parser);
-    instance.configureDefinition("[=unique]? [=modifier:immutable|settable|internal|defaulted|const]? [type,name>1,0] [[anonymous::attribute::1]]? ;");
-    
-    RulePart part = instance.getRulePart(2);
-    Assert.assertEquals("type,name>1,0",part.getName());
-    Assert.assertEquals(2, part.getNextIdentifiers().length);
-    Assert.assertEquals("=", part.getNextIdentifiers()[0]);
-    Assert.assertEquals(";", part.getNextIdentifiers()[1]);
-  }
-  
-  @Test
-  public void getNextIdentifier_EOLs()
-  {
-    parser.addRule("program- : [[comment]]* [[useStatement]]? [[comment]]*");
-    parser.addRule("comment- : [[inlineComment]] | [[multilineComment]]");
-    parser.addRule("useStatement- : use [useStatement] ;");
-    parser.addRule("inlineComment- : // [*inlineComment]");
-    parser.addRule("multilineComment- : /* [**multilineComment] */");
-
-    RuleInstance instance = new RuleInstance(parser);
-    instance.configureDefinition("[[comment]]* [[useStatement]]? [[comment]]*");
-    
-    RulePart part = instance.getRulePart(0);
-    Assert.assertEquals("comment",part.getName());
-    Assert.assertEquals(1, part.getNextIdentifiers().length);
-    Assert.assertEquals("use", part.getNextIdentifiers()[0]);
-  }  
-
-  @Test
-  public void getNextIdentifier_Nested()
-  {
-    parser.addRule("attribute : name [[value]] ;");
-    parser.addRule("value : [[start]] . [end]");
-    parser.addRule("start : xyz [front]");
-    
-    RuleInstance instance = new RuleInstance(parser);
-    instance.configureDefinition("name [[value]] ;");
-    
-    RulePart part = instance.getRulePart(0);
-    Assert.assertEquals("name",part.getName());
-    Assert.assertEquals(1,part.getNextIdentifiers().length);
-    Assert.assertEquals("xyz",part.getNextIdentifiers()[0]);
-  }
-
-}
+
+
+public class RuleInstanceTest
+{
+  
+  Parser parser;
+  
+  @Before
+  public void setUp()
+  {
+    parser = new Parser("p");
+  }
+
+  @Test
+  public void emptyConstructor()
+  {
+    RuleInstance instance = new RuleInstance(parser);
+    instance.configureDefinition("");
+    Assert.assertEquals(0,instance.numberOfRuleParts());
+  }
+  
+  @Test
+  public void staticConstructor()
+  {
+    RuleInstance instance = new RuleInstance(parser);
+    instance.configureDefinition("blah");
+    Assert.assertEquals(1,instance.numberOfRuleParts());
+    
+    RulePart part = instance.getRulePart(0);
+    Assert.assertEquals("blah",part.getName());
```

Comment:
A lot of the `Assert.assert`... could be cleaned up with a static import of `Assert`.